front/src/applications/operationalStudies/helpers/upsertMapWaypointsInOperationalPoints.ts

```diff
@@ -57,7 +63,7 @@ export const upsertMapWaypointsInOperationalPoints = (
           part: { track: step.track, position: step.offset },
           position: positionOnPath,
           weight: HIGHEST_PRIORITY_WEIGHT,
-        };
+        } as Point;
```

Comment:
I prefer to avoid casts. Here `Point` can be any type with extra properties, and these extra properties might be missing from the object we generate. It seems like typing is incorrect?

Why do we need a generic here in the first place?

Reply:
I confess I didn't like this cast either :sweat_smile: I had to add it, because otherwise TypeScript wasn't happy with a potential `NonNullable<PathProperties['operational_points']>[number]` with a `opId` property.

More precisely, here is the TypeScript error when I use `const formattedStep: Point = {...};` instead:

```
TS2322: Type
{
  id: string;
  opId: null;
  extensions: {
    identifier: {
      name: string;
      uic: number;
    };
  };
  part: {
    track: string;
    position: number;
  };
  position: number;
  weight: number;
}
is not assignable to type Point:
{
  id: string; 
  opId: null; 
  extensions: {
    identifier: {
      name: string; 
      uic: number; 
    }; 
  }; 
  part: {
    track: string; 
    position: number; 
  }; 
  position: number; 
  weight: number; 
}
is assignable to the constraint of type Point, but Point could be instantiated with a different subtype of constraint
{
  extensions?: OperationalPointExtensions | undefined;
  id: string;
  part: OperationalPointPart;
  position: number;
  weight: number | null;
} | PathOperationalPoint
```

@emersion How would you solve that, in that context?

Reply:
Would it be possible to just do : 
```suggestion
export function upsertMapWaypointsInOperationalPoints(
  path: TrainSchedule['path'],
  pathItemsPositions: PathfindingResultSuccess['path_item_positions'],
  operationalPoints: (NonNullable<PathProperties['operational_points']>[number] | PathOperationalPoint)[],
  t: TFunction<'operational-studies'>
): (NonNullable<PathProperties['operational_points']>[number] | PathOperationalPoint)[] {
```

And then 
```
const formattedStep: PathOperationalPoint = {
```